api/src/main/java/org/infinispan/api/search/reactive/ContinuousQueryPublisher.java

```diff
@@ -0,0 +1,26 @@
+package org.infinispan.api.search.reactive;
+
+import java.util.concurrent.Flow;
+import java.util.concurrent.TimeUnit;
+
+import org.infinispan.api.collections.reactive.KeyValueEntry;
+import org.reactivestreams.Publisher;
+
+public interface ContinuousQueryPublisher<K, V> extends Publisher<KeyValueEntry<K, V>>, Flow.Publisher<KeyValueEntry<K, V>> {
+
+   ContinuousQueryPublisher<K, V> query(String ickleQuery);
+
+   ContinuousQueryPublisher<K, V> withQueryParameter(String name, Object value);
+
+   ContinuousQueryPublisher<K, V> withQueryParameters(QueryParameters params);
+
+   ContinuousQueryPublisher<K, V> withTimeout(long timeout, TimeUnit timeUnit);
+
+   ContinuousQueryPublisher<K, V> created();
+
+   ContinuousQueryPublisher<K, V> updated();
+
+   ContinuousQueryPublisher<K, V> removed();
+
+   void dispose();
```

Comment:
I would recommend to hide this as an implementation detail. For users it is usually pain to manually release resources.

It could done automatically when the subscription is cancelled.

Besides, looking at the implementation, it seems that if you have multiple subscribers, you are going to stop emitting for all of them. This is not a standard behavior.

Reply:
thanks for this. I did not know how to handle this!